better-code/src/chapter-2-contracts.md

```diff
@@ -0,0 +1,1004 @@
+/// `areInOrder`.
+```
+
+Usually, the less our documentation looks like code (without
+sacrificing precision), the greater its power as a tool for
+understanding programs, writing tests, and verifying
+correctness. [^argument-order].
+
+[^argument-order]: One could argue that to not lose precision, the
+    documentation needs to say in which order the adjacent elements
+    are passed to `areInOrder`. Judgement calls are part of
+    engineering, and in our judgement the ordering is implied.
+
+There are two simplifications we can make to the precondition. First,
+everything in documentation of a method happens in the context of
+`self`, so we can drop “of `self`.”  This improvement was available to
+us all along, and may seem trivial, but the value of [omitting
+needless
+words](https://thewritinghabit.blog/2016/04/04/omit-needless-words/)
+adds up over lots of function definitions. It lowers the burden of
+writing documentation, and makes it much easier to read. Remember,
+every word a reader has to process has a cognitive cost in addition to
+whatever benefits it may provide, so each one should more than pay for
+itself.
+
+Secondly, the summary has become so simple that we can arguably embed
+the precondition there, making the final declaration:
+
+```swift
+/// Sorts the elements so that all adjacent pairs satisfy the [partial
+/// ordering](https://en.wikipedia.org/wiki/Partially_ordered_set)
+/// `areInOrder`.
+///
+/// - Complexity: at most N log N comparisons, where N is the number
+///   of elements.
+mutating func sort<T>(areInOrder: (T, T)->Bool) { ... }
+```
+
+There is one factor weighing against all these simplifications,
+though: sorting functions taking strict weak orderings are
+traditional. One has to weigh the risk of users reflexively writing
+`sort(areInOrder: <)`—against the value of these simplifications.  The
+argument label helps a bit, because *if* you happen to consider the
+equal elements case, you will see they clearly `areInOrder` but and
+realize `<` will return false.  But if you don't happen to consider
+that case, you won't get the sort you expected… sometimes.  Whether to
+break with precedent in order to get a simpler API with a clearer
+contract is an engineering decision you will have to make. To reduce
+the risk you could add this assertion, which will fire if the ordering
+is strict-weak:
+
+```
+assert(
+  self.isEmpty || areInOrder(first!, first!),
+  "Partial ordering required; did you pass a strict-weak ordering?")
+```
+
+
+### Project-Wide Documentation Policies
+
+The only reason you haven't seen
+complexity documented in this talk up to now is that I have a policy
+that operations have constant complexity unless specifically
+documented otherwise.
+
+Which brings me to this aside…  A big part of making the documentation
+problem tractable is having some well-chosen project-wide policies
+that save you from repeating common patterns. So I'm not going to
```

Comment:
You use "So" a lot in this chapter. I suggest using variations or omitting it so it doesn't stand out so much.